lib/platform/git/storage.ts

```diff
@@ -470,6 +463,10 @@ export class Storage {
     message,
   }: CommitFilesConfig): Promise<string | null> {
     logger.debug(`Committing files to branch ${branchName}`);
+    if (!this._config.privateKeySet) {
```

Comment:
Would set this directly to `this._privateKeySet` as this is no config option.

Why not directly write key on `setPrivateKey` call? Is it too expensive?

Reply:
Writing the private key takes three exec calls so I prefer to do is "lazily" and only write it if a commit is to be made. On the app it takes ~100ms to complete, so that would be 0.1s longer for the 99% of jobs with no new commits.